server/routes/updateGoal/updateGoalController.ts

```diff
@@ -10,25 +12,47 @@ export default class UpdateGoalController {
     const { prisonNumber, goalReference } = req.params
     const { prisonerSummary } = req.session
 
-    const actionPlan = await this.educationAndWorkPlanService.getActionPlan(prisonNumber, req.user.token)
-    if (actionPlan.problemRetrievingData) {
-      // There was a problem retrieving the prisoner's action plan
-      // TODO - RR-188
-      res.redirect('/error')
-      return
-    }
+    let updateGoalForm: UpdateGoalForm
+    if (req.session.updateGoalForm) {
+      updateGoalForm = req.session.updateGoalForm
+    } else {
+      const actionPlan = await this.educationAndWorkPlanService.getActionPlan(prisonNumber, req.user.token)
+      if (actionPlan.problemRetrievingData) {
+        // There was a problem retrieving the prisoner's action plan
+        // TODO - RR-188
+        res.redirect('/error')
+        return
+      }
+
+      const goalToUpdate = actionPlan.goals.find(goal => goal.goalReference === goalReference)
+      if (!goalToUpdate) {
+        // The requested goal is not part of the prisoners action plan
+        // TODO - RR-188
+        res.redirect('/error')
+        return
+      }
 
-    const goalToUpdate = actionPlan.goals.find(goal => goal.goalReference === goalReference)
-    if (!goalToUpdate) {
-      // The requested goal is not part of the prisoners action plan
-      // TODO - RR-188
-      res.redirect('/error')
-      return
+      updateGoalForm = toUpdateGoalForm(goalToUpdate)
     }
 
-    const updateGoalForm = toUpdateGoalForm(goalToUpdate)
+    req.session.updateGoalForm = undefined
 
     const view = new UpdateGoalView(prisonerSummary, updateGoalForm, req.flash('errors'))
     res.render('pages/goal/update/index', { ...view.renderArgs })
   }
+
+  submitUpdateGoalForm: RequestHandler = async (req, res, next): Promise<void> => {
+    const { prisonNumber, goalReference } = req.params
+    const updateGoalForm = { ...req.body }
+
+    const errors = validateUpdateGoalForm(updateGoalForm)
+    if (errors.length > 0) {
+      req.flash('errors', errors)
```

Comment:
If there are errors we put the form (with its blank/invalid fields) on the session, so that the `GET` handler above can use it (rather than reading the goal from the API again)